fineract-provider/src/main/java/org/apache/fineract/portfolio/client/api/ClientBusinessDetailApiResource.java

```diff
@@ -97,4 +98,36 @@ public String addBusinessDetail(@PathParam("clientId") final long clientId, fina
         return this.toApiJsonSerializer.serialize(result);
     }
 
+    @GET
+    @Path("{businessDetailId}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Retrieve Client Business Details by Id", description = "This is a convenience resource. It can be useful when building maintenance user interface screens for client business details applications. The data returned consists of any or all of:\n"
+            + "\n" + "Field Defaults\n" + "Allowed Value Lists\n\n" + "Example Request:\n" + "\n" + "clients/template")
+    public String retrieveBusinessDetail(@Context final UriInfo uriInfo, @PathParam("clientId") final long clientId,
+            @PathParam("businessDetailId") final long businessDetailId) {
+
+        this.context.authenticatedUser().validateHasReadPermission(ClientApiConstants.READ_CLIENTBUSINESSDETAIL);
+
+        ClientBusinessDetailData clientBusinessDetailData = null;
+        clientBusinessDetailData = this.businessDetailReadPlatformService.retrieveBusinessDetail(clientId, businessDetailId);
+
+        final ApiRequestJsonSerializationSettings settings = this.apiRequestParameterHelper.process(uriInfo.getQueryParameters());
+        return this.toApiJsonSerializer.serialize(settings, clientBusinessDetailData,
+                ClientApiConstants.CLIENT_BUSINESS_DETAIL_RESPONSE_REQUEST_DATA_PARAMETERS);
+    }
+
+    @DELETE
+    @Path("{businessDetailId}")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Delete a Client Business Detail", description = "Delete a Client Business Detail")
+    public String deleteCharge(@PathParam("businessDetailId") final Long businessDetailId, @PathParam("clientId") final Long clientId) {
```

Comment:
Shouldn't the method name be `deleteBusinessDetail`?

Reply:
yeah. Let me work on it. Thanks